internal/wasmldr/src/workload.rs

```diff
@@ -103,19 +109,46 @@ pub fn run<T: AsRef<str>, U: AsRef<str>>(
 
     debug!("creating WASI context");
     let mut wasi = WasiCtxBuilder::new();
-    for arg in args {
-        wasi = wasi.arg(arg.as_ref()).or(Err(Error::StringTableError))?;
-    }
-    for kv in envs {
-        wasi = wasi
-            .env(kv.0.as_ref(), kv.1.as_ref())
-            .or(Err(Error::StringTableError))?;
+
+    debug!("Processing loader config {:#?}", &ldr_config);
+
+    if let Some(ref files) = ldr_config.files {
+        for file in files.iter() {
```

Comment:
If the type is already a `&Vec` then in that case `for file in files` should work, since that should desugar to `for file in files.into_iter()` which should be equivalent to `for file in files.iter()` for `&Vec`: https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-IntoIterator

Reply:
Yes it does... don't ask me why I didn't write that, because I did the same loop just some lines below.